### DISCUSSION TITLE
Using the Matchmaking Addon

### DISCUSSION THREAD

**Discussion by Andrei965** [jeremylvln/Shulker#735] [Q&A] [answered]
Hello! [This](https://shulker.jeremylvln.fr/latest/guide/addons/matchmaking.html#creating-in-game-tickets) section of the wiki is marked as TODO. How can I create the ticket for matchmaking? What package do I need to include in my java project?

**Comment by jeremylvln** [comment, marked as answer]
Hi @Andrei965,

I've not dug the Matchmaking add-on since a really long time. I never took the time to finish it properly, but I was very close. If you are familiar with Open Match, you can probably find your path yourself, but the native integration with Shulker is very work-in-progress and not ready yet for use :(

**Reply by Andrei965** [reply]
Hi, thanks for your response! Could you provide some guidance on how I can create a game server from my code? Which package should I include in my project?

**Reply by Andrei965** [reply]
Hi @jeremylvln , it's been a while since we last spoke. I'm planning to set up a scalable minigame network using Minestom. My goal is to host multiple games on each Minestom server, each in its own isolated world, for greater efficiency. Could you provide some advice on how to achieve this with Shulker? Additionally, I'd like to know how I can develop my own matchmaking service and integrate it with Shulker.

**Reply by Andrei965** [reply]
Hi! I have tried to use the ShulkerSDK, but I can't figure out where to get the values for the env variables SHULKER_SDK_GRPC_HOST and SHULKER_SDK_GRPC_PORT

**Reply by Andrei965** [reply]
For anyone reading this later, you have to create your own matchmaker service and include the ShulkerSDK. You can get the env variables SHULKER_SDK_GRPC_HOST and SHULKER_SDK_GRPC_PORT from the shulker-system-shulker-operator service.

**Reply by jeremylvln** [reply]
Hi @Andrei965, sorry for the delay, quite busy, unfortunately. It's kinda hard to find time to work on Shulker.

The matchmaking engine is really WIP and I really want to try to abstract all the complexity of Open Match with a dead simple abstraction mechanism. I'm happy to hear that you managed to find the solution by yourself, do not hesitate to give me some feedbacks, I've not had the chance to try it myself yet.
